### PR TITLE
SOC-943 Post details on forum post listing

### PR DESCRIPTION
**Pull request by mklucsarits**
https://wikia-inc.atlassian.net/browse/HG-859 (https://wikia-inc.atlassian.net/browse/SOC-943)

These changes mainly move post-detail components onto the post list page, in order to summarize the details for each post. The PostDetailComponent was re-used from the main post page, and therefore, some CSS styling changes were needed as well outside the scope of just the post list page.

Additionally, this pull request has some changes to routing that were required to match the URL specification for Discussions.

### CODE REVIEW

**Comment by mklucsarits** [comment]
While merging 'dev' into this branch, there were a few conflicts. Be on the lookout for anything that doesn't look right, @garthwebb.

File: front/styles/main/module/_discussion.scss [hunk 1]
```diff
@@ -13,29 +14,27 @@
 		}
 	}
 
-	.see-more {
```
Comment (mklucsarits) [inline front/styles/main/module/_discussion.scss]: Some styles were moved in this file just to keep things sorted alphabetically.

File: front/templates/main/discussion/forum/index.hbs [hunk 2]
```diff
@@ -1,8 +0,0 @@
-<h1>Forum: {{name}}</h1>
```
Comment (mklucsarits) [inline front/templates/main/discussion/forum/index.hbs]: Moved to forum.hbs.

File: front/templates/main/discussion/post.hbs [hunk 3]
```diff
@@ -1,4 +1,4 @@
-<div class="discussion-post">
+<div class="discussion post">
```
Comment (mklucsarits) [inline front/templates/main/discussion/post.hbs]: This is intentional, to help define more granular CSS styles.

File: front/templates/main/discussion/post.hbs [hunk 4]
```diff
@@ -13,12 +13,12 @@
         </button>
 	{{/if}}
 	{{#each model.replies as |item index|}}
+		<hr>
```
Comment (lizlux) [inline front/templates/main/discussion/post.hbs]: I'm not a huge fane of `<hr>` tags b/c they are for baking display logic into HTML. A `border-bottom` css property would be better IMO.

Comment (lizlux) [inline front/templates/main/discussion/post.hbs]: I suppose if you use them for their semantic purpose of [paragraph-level thematic break](http://w3c.github.io/html-reference/hr.html) then they are okay, but I personally wouldn't use them.

Comment (mklucsarits) [inline front/templates/main/discussion/post.hbs]: Yup, good point. I'll ditch them for borders.

File: front/scripts/main/controllers/DiscussionForumPostController.ts [hunk 5]
```diff
@@ -1,20 +0,0 @@
-/// <reference path="../app.ts" />
```
Comment (mklucsarits) [inline front/scripts/main/controllers/DiscussionForumPostController.ts]: All logic here has been moved to DiscussionPostRoute.ts.

File: front/scripts/main/models/DiscussionPostModel.ts [hunk 6]
```diff
@@ -44,25 +44,28 @@ App.PostModel = Em.Object.extend({
...
 			Em.$.ajax({
-				url: `https://services.wikia-dev.com/discussion/${wikiId}/threads/${threadId}` +
+				url: `https://services.wikia.com/discussion/${wikiId}/threads/${threadId}` +
 					 '?responseGroup=full&sortDirection=descending&limit=' + postInstance.replyLimit,
-
 				dataType: 'json',
 				success: (data: any) => {
 					var replies = data._embedded['doc:posts'],
-						pivotId = replies[0].id;
+						pivotId;
```
Comment (mklucsarits) [inline front/scripts/main/models/DiscussionPostModel.ts]: These few lines are out of scope of this ticket, but needed to be fixed due to a change in the API response for posts without replies.

File: front/scripts/main/routes/DiscussionForumPostRoute.ts [hunk 7]
```diff
@@ -1,17 +0,0 @@
-/// <reference path="../app.ts" />
```
Comment (mklucsarits) [inline front/scripts/main/routes/DiscussionForumPostRoute.ts]: This file was renamed to DiscussionPostRoute.ts, due to routing changes and for brevity.

File: front/scripts/main/routes/DiscussionPostRoute.ts [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+		this.controllerFor('application').set('themeBar', true);
+	},
+
+	deactivate (): void {
+		// Disables vertical-colored theme bar in site-head component
+		this.controllerFor('application').set('themeBar', false);
+	},
+
+	showMore: Em.computed('model', function (): boolean {
+			var model = this.get('model'),
```
Comment (rogatty) [inline front/scripts/main/routes/DiscussionPostRoute.ts]: Are you sure this works? In a route `this.model` is a function, look at line 4. You can use `this.modelFor('discussion.post')` to get the actual model.

**Comment by garthwebb** [comment]
LGTM

File: front/scripts/main/routes/DiscussionPostRoute.ts [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+
+			return loadedRepliesLength < model.postCount;
+	}),
+
+	actions: {
+		expand: function () {
+			var model = this.modelFor('discussion.post');
+
+			model.loadNextPage();
+		},
```
Comment (lizlux) [inline front/scripts/main/routes/DiscussionPostRoute.ts]: extra comma

File: front/templates/main/discussion/forum/index.hbs (hunk 2 above)
Comment (lizlux) [inline front/templates/main/discussion/forum/index.hbs]: Could these file changes have been done with `git mv`?

**Comment by lizlux** [comment]
Once the comments are addressed, this LGTM.

File: front/templates/main/discussion/forum/index.hbs (hunk 2 above)
Comment (mklucsarits) [inline front/templates/main/discussion/forum/index.hbs]: They actually were. I think if you make changes to the file after it's moved, Github won't show that it was moved.

**Pull request merged**